stdlib/public/Concurrency/Task.swift

```diff
@@ -103,11 +88,14 @@ extension Task {
       do {
         return .success(try await value)
       } catch {
-        return .failure(error as! Failure) // as!-safe, guaranteed to be Failure
+        return .failure(error)
```

Comment:
The `error as! Failure` cast isn't needed in this PR, but if `Failure != Error` is it unsafe for the runtime to throw a `CancellationError` when cancelled "externally"?

```swift
  /// ### Cancellation
  /// If the awaited on task gets cancelled externally the `get()` will throw
  /// a cancellation error.
```